### ISSUE TITLE
Code is broken for wasm32-unknow-emscripten

### ISSUE THREAD

**Issue by caiiiycuk** [ron-rs/ron#335]
When I try to build project that depends on ron@0.7.0  for target wasm32-unknow-emscripten, I have lot of messages like this:
```
  --> /home/caiiiycuk/.cargo/registry/src/github.com-1ecc6299db9ec823/ron-0.7.0/src/de/id.rs:74:5
   |
74 | /     fn deserialize_i128<V>(self, _: V) -> Result<V::Value>
75 | |     where
76 | |         V: Visitor<'b>,
77 | |     {
78 | |         unimplemented!("IdDeserializer may only be used for identifiers")
79 | |     }
   | |_____^ not a member of trait `de::Deserializer`
```

I believe it's because emscripten target does not have i128/u182 types. Serde in that case does not have deserialize_i128/u128. As workaround I use version 0.5.1, it does not contains i128/u128 and works fine. However I would like to update to latest version if possible.

**Comment by MomoLangenstein** [comment]
Related to #304

**Comment by MomoLangenstein** [comment]
I've done a tiny bit of investigation. As `serde` only provides [serde_if_integer128](https://docs.serde.rs/serde/macro.serde_if_integer128.html) but no else variant, I did not find a way to tap into just `serde` to e.g. change the definition of https://github.com/ron-rs/ron/blob/7fefc00930c83f40b767a584cc2e4441c3e27d95/src/parse.rs#L89-L103. So I think `#[cfg()]`s are needed. While `ron` could mirror part of `serde`'s build script, this does seem overkill. Adding a separate `integer128` feature as suggested in #304 seems like the better option. This could then be tested by trying to build with target `wasm32-unknown-emscripten` on `1.36.0`. Or `ron` could bump its MSRV to `1.40` which is when emscripten gained support for 128b integers. @caiiiycuk what Rust version are you using?

**Comment by caiiiycuk** [comment]
1.53.0

**Comment by torkleyy** [comment]
I think adding wasm to the build matrix makes sense. @kvark thoughts on this? I think a feature for `integer128` would make sense, most of the time you don't need `i128` anyways IMO. I see that features increase the complexity when it comes to testing and build scripts but this seems like a good reason.

**Comment by kvark** [comment]
Yes, it all sounds reasonable. Although, I think @caiiiycuk 's original need for this issue is no longer active, if I understand correctly. This was needed for vange-rs, and the current plan is to use `wasm32-unknown-unknown` instead.

**Comment by caiiiycuk** [comment]
Correct, but I think for consistency should be fixed at some time. 
+1 for adding feature

**Comment by MomoLangenstein** [comment]
@caiiiycuk Has this been fixed with #351?

**Comment by caiiiycuk** [comment]
Just checked on vange-rs, works like a charm!
